tests/common/fixtures/duthost_utils.py

```diff
@@ -3,17 +3,36 @@
 
 logger = logging.getLogger(__name__)
 
-@pytest.fixture
-def backup_and_restore_config_db(duthost):
-    """
-    Some cases will shutdown interfaces or BGP in test, and the change is writen to
-    config db after warm-reboot. Therefore, we need to backup config_db.json before
-    test starts and then restore after test ends
+
+def _backup_and_restore_config_db(duthost):
+    """Back up the existing config_db.json file and restore it once the test ends.
+
+    Some cases will update the running config during the test and save the config
+    to be recovered aftet reboot. In such a case we need to backup config_db.json before
+    the test starts and then restore it after the test ends.
     """
     CONFIG_DB = "/etc/sonic/config_db.json"
     CONFIG_DB_BAK = "/etc/sonic/config_db.json.before_test"
     logger.info("Backup {} to {}".format(CONFIG_DB, CONFIG_DB_BAK))
     duthost.shell("cp {} {}".format(CONFIG_DB, CONFIG_DB_BAK))
+
     yield
+
     logger.info("Restore {} with {}".format(CONFIG_DB, CONFIG_DB_BAK))
     duthost.shell("mv {} {}".format(CONFIG_DB_BAK, CONFIG_DB))
+
+
+@pytest.fixture
+def backup_and_restore_config_db(duthost):
```

Comment:
@bingwang-ms can you help me verify that this works OK for the advanced_reboot tests? It seemed fine to me but I think a second set of eyes would be helpful. :)